doc/DesignNotes.md

```diff
@@ -21,5 +21,24 @@ Database access layer is defined in `org.ergoplatform.explorer.db` module and ha
     - `composite` - module containing models which are not mapped to a single db table exactly but consists of a data from several tables
 
 - `queries` - module containing sets of sql queries for each entity for concrete database
-- `repositories` - defines abstract interfaces for accessing entities' data and also their production implementations using `queries` currently.
-````
+- `repositories` - defines abstract interfaces for accessing entities' data and also their production implementations.
+- `services` - defines sets of operations on different domain areas.
+- `protocol`
+    - `models` - models mirroring structure of corresponding objects returned by ergo node REST API. 
+
+#### Chain Grabber
+
+Chain grabber is a module responsible for blockchain synchronization between network and local db, it runs as a separate process and
+operates according to the following workflow:
+
+- Get height of the best block in the network
+- Compare it with the height of the best block in the local db
+- Request block ids at each height from local height to network height
+- Request full block for each id
+- Update chain statuses of existing blocks at this height (in case of fork processing)
+- For each new full block fetched from the network:
+    - Lookup its parent info in the cache or in the database
+        + If parent is not found fetch ids of all existing blocks at previous height and grab new blocks from it
+        + Otherwise split API block into separate db models described above and insert the to the db
+        
+The key feature of grabber is that it performs chain update at either particular height or range of heights (in case of fork processing) atomically.
```

Comment:
Single block at height is meant here. But that is not the only case -- when processing fork we need to update the whole range of heights affected by the fork, otherwise chain representation in DB can appear inconsistent for some time.
So your assumption about "append-only" blockchain state nature is incorrect.

Reply:
Will make description more clear.

Reply:
Fixed